### PR TITLE
make lube speed up lathes

### PR DESCRIPTION
**Pull request by deltanedas**
## About the PR
title

pour in lube and it makes items 4x faster, 5u per recipe

pour in glue it and makes items 5x slower :trollface:

if you poured in non lube you can spill it on the floor :trollface:

## Why / Balance
real industrial use for lube as stated in its description

sci makes big stockpile of lube for mass producing shit (especially to counter hyperlathes slowdown) and clown plans a heist :trollface:

## Technical details
ReagentSpeedComponent added which maps reagents to speed modifiers, system has ApplySpeed which calculates the final speedup or slowdown

combines glue and lube at once properly

## Media

no lube

yes lube

fast

https://github.com/space-wizards/space-station-14/assets/39013340/fc34bff7-950c-41ff-933c-ffa63bedfa62

glue work trust

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes
no

**Changelog**
:cl:
- tweak: Lathes can now be sped up by using Space Lube.

### CODE REVIEW

**Comment by IamVelcroboy** [comment]
This is a really good idea.

> clown breaks into sci and pours in glue, slows down the lathe

and, honestly, so is this 😄

**Comment by ViceEmargo** [comment]
Clowns are in shambles after scientists have already raided maintenance looking for space lube bottles.

**Comment by deltanedas** [comment]
clowns then adding $10k lube tanks to cargo, for express delivery to science of course and not for bringing to the clown room

**Comment by Kadeo64** [comment]
why would you demonstrate this with a video of a circuit imprinter instead of a video of a hyperconvection lathe printing 400 svalinns

**Comment by deltanedas** [comment]
shitsci didnt research it :trollface:

**Comment by Kadeo64** [comment]
> shitsci didnt research it :trollface:

wouldn't it be more shitsci if they did research it 🤔

**Comment by deltanedas** [comment]
shitsci researching a single arsenal tech :trollface:

**Comment by FairlySadPanda** [comment]
Should this be combined with a slowing down of the lathes by default?

**Comment by deltanedas** [comment]
> slowing down of the lathes by default

how it is right now it wouldn't be entirely required, if normal lathes are slower by default then its a basic necessity

**Comment by Ghagliiarghii** [comment]
I like this idea and also the one about glue slowing it but I feel the speed up is too dramatic. Should be more like the speed up it had with Advanced Manipulators. (Also I don't support slowing it down so it requires space lube to be equivalent to how it is now)

**Comment by deltanedas** [comment]
the 4x speedup is a bit insane yes but due to limited chems you have to actually decide when you want the speedup instead of leaving it in forever

**Comment by Emisse** [comment]
feels a bit odd to make a lathelubricantcomponent with hardcoded prototype ids instead of a generic lubricant component that you could apply to change a float on lathe speed so you could add to glue and do like 0.2 or something. JUST MY THOUGHTS

**Comment by deltanedas** [comment]
i dont think thats possible, unless individual reagents become entities then raise event on it and lube/glue/water/whatever handle :trollface:

i will try make it not lathe specific however then the event with be lube specific

**Comment by deltanedas** [comment]
@EmoGarbage404 trolling complete

**Comment by EmoGarbage404** [comment]
the hyper convection lathes either need to have the component removed or have a less effective but more costly lubricant. They are already extremely powerful and with the heat generation tied to the amount of time it takes to print, speeding up print time could make them even more of a non-issue.

**Comment by deltanedas** [comment]
> with the heat generation tied to the amount of time it takes to print

i thought heat was just per-recipe, making it have less effect on hyperlathes then

File: Resources/Prototypes/Entities/Structures/Machines/lathe.yml [hunk 1]
```diff
@@ -231,16 +244,12 @@
 
 - type: entity
   id: AutolatheHyperConvection
-  parent: Autolathe
+  parent: [Autolathe, BaseHyperlathe]
```
Comment (EmoGarbage404) [inline Resources/Prototypes/Entities/Structures/Machines/lathe.yml:247]: does it not inherit the component definition from `Autolathe` first, before getting the one from `BaseHyperLathe`

Comment (deltanedas) [reply Resources/Prototypes/Entities/Structures/Machines/lathe.yml:247]: i think its the opposite??? will need to test

Comment (deltanedas) [reply Resources/Prototypes/Entities/Structures/Machines/lathe.yml:247]: test soon i forgor

Comment (deltanedas) [reply Resources/Prototypes/Entities/Structures/Machines/lathe.yml:247]: no im right BaseHyperlathe comes after, the resulting prototype has 0.5 usage 1.5 time as expected

**Comment by deltanedas** [comment]
@EmoGarbage404 merg

**Pull request merged**